### PR TITLE
sap_hypervisor_node_preconfigure: Security fixes regarding assert|that

### PR DESCRIPTION
**Pull request by newkit**
Adressed fixes required due to issues with assert | that statements when using curled brackets. 
Tested the role and fixed minor issues as well.

### CODE REVIEW

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_rhel_kvm/assert-configuration.yml [hunk 1]
```diff
@@ -70,26 +70,29 @@
       ignore_errors: "{{ sap_hypervisor_node_preconfigure_ignore_failed_assertion }}"
 
 - name: Assert - check sap_hypervisor_node_preconfigure_nx_huge_pages
-  when: sap_hypervisor_node_preconfigure_nx_huge_pages is defined
+  when: sap_hypervisor_node_preconfigure_kvm_nx_huge_pages is defined
   block:
-    - name: "Assert - Check kvm.nx_huge_pages is {{ sap_hypervisor_node_preconfigure_nx_huge_pages }}"
+    - name: Set fact for sap_hypervisor_node_preconfigure_register_assert_nx_huge_pages
+      ansible.builtin.set_fact:
```
Comment (Wabri) [inline roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_rhel_kvm/assert-configuration.yml:76]:
```suggestion
      ansible.builtin.set_fact:
```

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_rhel_kvm/assert-configuration.yml [hunk 2]
```diff
@@ -70,26 +70,29 @@
...
-    - name: "Assert - Check kvm.nx_huge_pages is {{ sap_hypervisor_node_preconfigure_nx_huge_pages }}"
+    - name: Set fact for sap_hypervisor_node_preconfigure_register_assert_nx_huge_pages
+      ansible.builtin.set_fact: 
+        sap_hypervisor_node_preconfigure_register_assert_nx_huge_pages: "{{ __sap_hypervisor_node_preconfigure_kernelcmdline_assert.stdout | regex_search('kvm.nx_huge_pages=(.+)', '\\1') | first }}"
+    - name: "Assert - Check kvm.nx_huge_pages is {{ sap_hypervisor_node_preconfigure_kvm_nx_huge_pages }}"
       ansible.builtin.assert:
-        that: "'kvm.nx_huge_pages={{ sap_hypervisor_node_preconfigure_nx_huge_pages }}' in __sap_hypervisor_node_preconfigure_kernelcmdline_assert.stdout"
-        fail_msg: "FAIL: kvm.nx_huge_pages is not {{ sap_hypervisor_node_preconfigure_nx_huge_pages }}"
-        success_msg: "PASS: kvm.nx_huge_pages is {{ sap_hypervisor_node_preconfigure_nx_huge_pages }}"
+        that: sap_hypervisor_node_preconfigure_register_assert_nx_huge_pages == sap_hypervisor_node_preconfigure_kvm_nx_huge_pages
```
Comment (Wabri) [inline roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_rhel_kvm/assert-configuration.yml:80]:
```suggestion
        that: sap_hypervisor_node_preconfigure_register_assert_nx_huge_pages == sap_hypervisor_node_preconfigure_kvm_nx_huge_pages
```

**Review by Wabri — CHANGES_REQUESTED** [review]
2 minimal necessary changes in order to pass the ansible-lint workflow

**Comment by newkit** [comment]
Thanks @Wabri, I've made the two changes requested. Was pushing a bit too fast before checking for the lint results *ducks*

**Comment by Wabri** [comment]
Awesome ⚡

**Review by newkit — COMMENTED** [review]
Addressed commented issues about trailing spaces.

**Review by berndfinger — COMMENTED** [review]
LGTM

**Review by sean-freeman — APPROVED** [review]
lgtm 👍

**Review by berndfinger — APPROVED** [review]

**Comment by sean-freeman** [comment]
@Wabri please confirm you are happy your changes have been met by providing `Approve` and `lgtm :+1:` when ready

**Comment by newkit** [comment]
Confirm Wabri's review and requested changes have been met, this PR is ready to merge

**Comment by sean-freeman** [comment]
OK I can see @Wabri confirmed above. Will merge given the changes requested were trailing spaces.

**Pull request merged**